docs/guides/authentication.md

```diff
@@ -48,3 +48,69 @@ class Mutation:
 
         return LoginSuccess(user=User(username=username))
 ```
+
+### Access authenticated user in resolver
+
+Its fairly common to require user information within a resolver. We can do that in a type safe way with a custom context dataclass.
+
+For example, in fastapi this might look like this:
+
+```python
+from functools import cached_property
+from typing import cast
+
+import strawberry
+from fastapi import FastAPI, Request
+from strawberry.fastapi import BaseContext, GraphQLRouter
+from strawberry.types import Info as _Info
+from strawberry.types.info import RootValueType
+
+
+class User:
+    pass
+
+
+class ProgrammingError(Exception):
+    pass
+
+
+class Context(BaseContext):
```

Comment:
Wondering if it makes sense to extend the `BaseContext` here, or just inline the fields, so they are not optional for the usage later.

Reply:
@Meemaw I would remove the `_request` property and just check for `.request` inside the `user` property function. I think that makes the example a bit simpler.

Reply:
Its optional, so they we need to handle that there.

Reply:
I get that but it in the context of a simple example to explain the concept it adds unnecessary redirection IMO.

I think the `user` property should look something like this:

```python
@property
def user(self) -> Optional[User]:
    if not self.request:
        return None

    authorization_key = self._request.headers.get("Authorization", None)
    return authorization_service.authorize(authorization_key)
```

By making the User return optional it handles the case where the request is unauthenticated as well.

Oh and it should **not** be a `cached_property` because that cache will persist across requests.

Reply:
You might call this a lot of times in resolvers (especially within connections). You don't want to do the authorization logic each time. AFAIK `cached_property` is scoped to the current class instance, in this case `Context`  which gets recreated with every request.

Reply:
> AFAIK cached_property is scoped to the current class instance, in this case Context  which gets recreated with every request.

My bad, you're right. cached_property makes sense here then.